### ISSUE TITLE
Fix test framework for #688

### ISSUE THREAD

**Issue by ono-max** [ruby/debug#689]
https://github.com/ruby/debug/pull/688

**Comment by ono-max** [comment]
I'll fix it later.

**Comment by st0012** [comment]
For the DAP test, the fix is the same as: https://github.com/ruby/debug/pull/640/files#diff-edcbc39076c34506ea43bae159c4b26104156861edc7b84224e85cc4e980d355

**Comment by ono-max** [comment]
Oh, I see.

Could you create the PR instead of me?

**Comment by ono-max** [comment]
Thank you for asking me. I think that more test cases are better than less test cases, though?

Although I wrote the raw test in https://github.com/ruby/debug/pull/705, I'll close it if you disagree with it.

**Comment by st0012** [comment]
> I think that more test cases are better than less test cases, though?

That's true but we also need to consider the maintenance cost. If you look at #705, you can see that the test content may have been outdated for a while. And that makes me think:

- If we go stricter on matching the req/res, it'll need to be updated more frequently. But until now the missed changes don't seem to cause issues, so perhaps it's fine to keep it this way?
- If we decide that a rough behavior match like the current one is good enough, why don't we just maintain the request tests?

I of course appreciate your work on #705 so I don't think we should close it. But if the raw tests in the current form are worth maintaining is debatable imo.

**Comment by st0012** [comment]
How about this: if you can list one or more scenarios that a raw test case covers that its request counterpart doesn't, we keep it.  In addition to that, we should also document it done in the file so it'll help us maintain it.
If not, we remove it.
